iree/turbine/kernel/wave/wave.py

```diff
@@ -521,23 +519,30 @@ def test_execute(self, args, kwargs):
         config = kwargs.get("run_config", None)
         use_scheduling = kwargs.get("schedule", False)
         use_scheduling_barriers = kwargs.get("use_scheduling_barriers", False)
+        # When this is passed in from the user, we will populate it with the kernel hash.
+        cached_kernel_hash = kwargs.get("kernel_hash", [])
 
         # Get cached kernel when available.
         cache_enabled = is_cache_enabled()
+        kernel_hash = None
         if cache_enabled:
             cache_manager = get_cache_manager()
             # TODO: Move use_scheduling, use_scheduling_barriers, etc. into the config so everything is contained there.
-            processed_constraints = anonymize_constraints(self.constraints)
-            kernel_hash = cache_manager.get_hash(
-                processed_constraints,
-                self._f,
-                safe_dict_to_tuple(IndexingContext.current().subs),
-                tuple(dynamic_symbols),
-                safe_dict_to_tuple(config),
-                use_scheduling=use_scheduling,
-                use_scheduling_barriers=use_scheduling_barriers,
-                run_bench=run_bench,
-            )
+            if cached_kernel_hash:
+                kernel_hash = cached_kernel_hash[0]
```

Comment:
I don't understand why are we always reutrning the first index in cached_kernel_hash? Shouldn't we do some index check or something?

Reply:
So the idea is that when you use this flag, you are confident that you have only one kernel it should be cached and thats why I only load the first element. But open to suggestions

Reply:
I have modified this so that the length of cached_kernel_hash will always be 1.